webhooks-extension/pkg/endpoints/sink.go

```diff
@@ -227,6 +230,99 @@ func createPipelineRunFromWebhookData(buildInformation BuildInformation, r Resou
 	logging.Log.Debugf("PipelineRun created: %+v.", pipelineRun)
 }
 
+// This is the task creation for the minitoring the main pipelinerun and reporting the resule to the github
+func createTaskRunFromWebhookData(buildInformation BuildInformation, r Resource) {
+	logging.Log.Debugf("In createTaskRunFromWebhookData, build information: %s.", buildInformation)
+
+	// Install namespace
+	installNs := os.Getenv("INSTALLED_NAMESPACE")
+	if installNs == "" {
+		installNs = "default"
+	}
+
+	logging.Log.Debugf("Looking for the pipeline configmap in the install namespace %s.", installNs)
+
+	// get information from related githubsource instance
+	webhook, err := r.getGitHubWebhook(buildInformation.REPOURL, installNs)
+	if err != nil {
+		logging.Log.Errorf("error getting github webhook: %s.", err.Error())
+		return
+	}
+
+	taskTemplateName := webhook.PullTask
+	taskNs := webhook.Namespace
+	saName := webhook.ServiceAccount
+	accessTokenRef := webhook.AccessTokenRef
+	pullTaskParam1 := webhook.PullTaskParam1
+	pullTaskParam2 := webhook.PullTaskParam2
+
+	if saName == "" {
+		saName = "default"
+	}
+
+	if taskTemplateName == "" {
+		taskTemplateName = "monitor-result-task"
+	}
+
+	if pullTaskParam1 == "" {
+		pullTaskParam1 = "OK"
+	}
+
+	if pullTaskParam2 == "" {
+		pullTaskParam2 = "ERROR"
+	}
+
+	logging.Log.Debugf("Build information: %+v.", buildInformation)
+
+	// Assumes you've already applied the yml: so the task difinition must exist upfront.
+	startTime := buildInformation.TIMESTAMP
+	generatedPipelineRunName := fmt.Sprintf("%s-%s", webhook.Name, startTime)
+	generatedTaskRunName := generatedPipelineRunName
+
+	// Unique names are required so timestamp them.
```

Comment:
There's loads of duplicated code here, do we want the branch name in as well (I recently added it for our generated PipelineRuns)? Can anything be extracted into methods to reduce duplication?

Reply:
I looked through these 2 methods.  They look similar but they have only a little duplication.  I would like to keep them this way.